### PR TITLE
Kernel/FileSystem: Re-organize all filesystem implementations into separate files

### PR DESCRIPTION
**Pull request by supercomputer7**
No functional changes, just making it a bit more easy to read the code, also removed a bunch of accumulated unnecessary headers being included in the code.

This PR makes the directory structure to resemble the directory code structure in Linux now, where in the `fs` directory there are bunch of filesystem implementations, each in its own sub-directory to ensure it is more convenient to read the source code, as well as to change it.

### CODE REVIEW

**Comment by ADKaster** [comment]
Does this PR conflict with #15593 ? if so i'd rather that one get fixed up and merged before reviewing this one.

**Comment by supercomputer7** [comment]
> Does this PR conflict with #15593 ? if so i'd rather that one get fixed up and merged before reviewing this one.

It does conflict with it… that PR already has conflicts and seems like it gets no further response, so I will encourage to start reviewing this…

File: Meta/run.sh [hunk 1]
```diff
@@ -280,6 +280,7 @@ if [ -z "$SERENITY_MACHINE" ]; then
         -device i82801b11-bridge,id=bridge3 -device sdhci-pci,bus=bridge3
         -device ich9-ahci,bus=bridge3
         -chardev stdio,id=stdout,mux=on
+        -monitor vc
```
Comment (bgianfo) [inline Meta/run.sh:283]: Unintended Change?

File: Kernel/FileSystem/SysFS/DirectoryInode.cpp [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    // NOTE: No locking required as m_associated_component or its component index will never change during our lifetime.
+    InodeMetadata metadata;
+    metadata.inode = { fsid(), m_associated_component->component_index() };
+    metadata.mode = S_IFDIR | S_IRUSR | S_IRGRP | S_IROTH | S_IXOTH;
+    metadata.uid = 0;
+    metadata.gid = 0;
+    metadata.size = 0;
+    metadata.mtime = TimeManagement::boot_time();
+    return metadata;
+}
```
Comment (bgianfo) [inline Kernel/FileSystem/SysFS/DirectoryInode.cpp:37]: Shouldn't clang format have inserted a new line after this `}`?

Comment (supercomputer7) [reply Kernel/FileSystem/SysFS/DirectoryInode.cpp:37]: Fixed. I don't know why clang-format didn't fix it though.

File: Kernel/FileSystem/ProcFS/DirectoryInode.cpp [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    MutexLocker locker(m_inode_lock);
+    InodeMetadata metadata;
+    metadata.inode = { fsid(), m_associated_component->component_index() };
+    metadata.mode = S_IFDIR | m_associated_component->required_mode();
+    metadata.uid = m_associated_component->owner_user();
+    metadata.gid = m_associated_component->owner_group();
+    metadata.size = 0;
+    metadata.mtime = m_associated_component->modified_time();
+    return metadata;
+}
```
Comment (ADKaster) [inline Kernel/FileSystem/ProcFS/DirectoryInode.cpp:35]: Looks like another missing newline here, to continue brian's nitpick

File: Kernel/FileSystem/ProcFS/ProcessPropertyInode.cpp [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+    return adopt_nonnull_lock_ref_or_enomem(new (nothrow) ProcFSProcessPropertyInode(procfs, file_description_index, pid));
+}
+ErrorOr<NonnullLockRefPtr<ProcFSProcessPropertyInode>> ProcFSProcessPropertyInode::try_create_for_thread_stack(ProcFS const& procfs, ThreadID stack_thread_index, ProcessID pid)
+{
+    return adopt_nonnull_lock_ref_or_enomem(new (nothrow) ProcFSProcessPropertyInode(procfs, stack_thread_index, pid));
+}
+ErrorOr<NonnullLockRefPtr<ProcFSProcessPropertyInode>> ProcFSProcessPropertyInode::try_create_for_pid_property(ProcFS const& procfs, SegmentedProcFSIndex::MainProcessProperty main_property_type, ProcessID pid)
+{
+    return adopt_nonnull_lock_ref_or_enomem(new (nothrow) ProcFSProcessPropertyInode(procfs, main_property_type, pid));
+}
```
Comment (ADKaster) [inline Kernel/FileSystem/ProcFS/ProcessPropertyInode.cpp:29]: Newlines in this file too, basically the whole file. Seems we're quite inconsistent in the kernel about this lint rule :/

File: Kernel/FileSystem/ext2_fs.h [hunk 5]
```diff
@@ -1,733 +0,0 @@
...
- * Copyright (C) 1992, 1993, 1994, 1995
- * Remy Card (card@masi.ibp.fr)
- * Laboratoire MASI - Institut Blaise Pascal
- * Universite Pierre et Marie Curie (Paris VI)
- *
- *  from
- *
- *  linux/include/linux/minix_fs.h
- *
- *  Copyright (C) 1991, 1992  Linus Torvalds
```
Comment (ADKaster) [inline Kernel/FileSystem/ext2_fs.h:13]: I don't think we want to re-organize this file out into our own custom definitions file, it's GPL iirc. We should keep it as-is. Or at least, reimplement it completely while looking only at an ext2 spec document. I think that should be done in a separate PR.

File: Meta/check-style.py [hunk 6]
```diff
@@ -44,8 +44,6 @@ def should_check_file(filename):
         return False
     if filename.startswith('Base/'):
         return False
-    if filename == 'Kernel/FileSystem/ext2_fs.h':
```
Comment (ADKaster) [inline Meta/check-style.py:47]: and keep the exceptions in Meta, unless we go full read-spec-only redefintion of the ext2 constants and structures.

**Review by ADKaster — CHANGES_REQUESTED** [review]
Overall seems like reasonable reorganization. I'm a bit worried about the ext2_fs.h file though.

File: Kernel/FileSystem/ext2_fs.h (hunk 5 above)
Comment (supercomputer7) [reply Kernel/FileSystem/ext2_fs.h:13]: It will not pass CI otherwise... or maybe it's just that I need to update the lint scripts to not look into this updated location.

File: Kernel/FileSystem/SysFS/LinkInode.h [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <Kernel/FileSystem/SysFS/Inode.h>
+
+namespace Kernel {
+
+class SysFSLinkInode : public SysFSInode {
```
[the remark this answers is not in the document]
Comment (supercomputer7) [reply Kernel/FileSystem/SysFS/LinkInode.h:13]: Not so sure about this idea, we will have to discuss this in a separate method of communications, such as the Discord server.

**Review by ADKaster — APPROVED** [review]
